xml/telnet_banners.xml

```diff
@@ -0,0 +1,735 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<fingerprints protocol="telnet" database_type="service" preference=".80">
+  <!--
+  TELNET banners with CR/LF/whitespace trimmed from either end.
+  Examples with CR, LF, etc must be base64 encoded in order to past tests.
+  Please follow the style established below.
+  -->
+  <fingerprint pattern="^(?:\r|\n)*MikroTik v([\w.]+)(?: \(\w+\))?(?:\r|\n)+Login:\s*$">
+    <description>MikroTik RouterOS</description>
+    <!-- MikroTik v5.2\r\nLogin: -->
+    <example _encoding="base64" os.version="5.2">TWlrcm9UaWsgdjUuMg0KTG9naW46Cg==</example>
+    <!-- MikroTik v6.42.3 (stable)\r\nLogin: -->
+    <example _encoding="base64" os.version="6.42.3">TWlrcm9UaWsgdjYuNDIuMyAoc3RhYmxlKQ0KTG9naW46Cg==</example>
+    <!-- MikroTik v6.40.8 (bugfix)\r\nLogin: -->
+    <example _encoding="base64" os.version="6.40.8">TWlrcm9UaWsgdjYuNDAuOCAoYnVnZml4KQ0KTG9naW46Cg==</example>
+    <!-- MikroTik v6.36rc12 (testing)\r\nLogin: -->
+    <example _encoding="base64" os.version="6.36rc12">TWlrcm9UaWsgdjYuMzZyYzEyICh0ZXN0aW5nKQ0KTG9naW46Cg==</example>
+    <param pos="0" name="service.vendor" value="MikroTik"/>
+    <param pos="0" name="service.product" value="telnet"/>
+    <param pos="0" name="os.vendor" value="MikroTik"/>
+    <param pos="0" name="os.device" value="Router"/>
+    <param pos="0" name="os.product" value="RouterOS"/>
+    <param pos="1" name="os.version"/>
+    <param pos="0" name="os.cpe23" value="cpe:/o:mikrotik:routeros:{os.version}"/>
+    <param pos="0" name="hw.vendor" value="MikroTik"/>
+    <param pos="0" name="hw.device" value="Router"/>
+  </fingerprint>
+  <fingerprint pattern="^(?:\r|\n)?ZXHN (\w+)(?: V([\d.]+))?(?:\r|\n)*Login:\s*$">
+    <description>ZTE ZXHN router</description>
+    <!-- ZXHN H108N\r\nLogin: -->
+    <example _encoding="base64" hw.product="H108N">WlhITiBIMTA4Tg0KTG9naW46Cg==</example>
+    <!-- ZXHN H298A V1.1\r\nLogin: -->
+    <example _encoding="base64"  hw.product="H298A" hw.version="1.1">WlhITiBIMjk4QSBWMS4xDQpMb2dpbjoK</example>
+    <!-- ZXHN H367N\r\n\rLogin: -->
+    <example _encoding="base64" hw.product="H367N">WlhITiBIMzY3Tg0KDUxvZ2luOgo=</example>
+    <param pos="0" name="service.vendor" value="ZTE"/>
+    <param pos="0" name="service.product" value="telnet"/>
+    <param pos="0" name="hw.vendor" value="ZTE"/>
+    <param pos="0" name="hw.device" value="Router"/>
+    <param pos="0" name="hw.family" value="ZXHN"/>
+    <param pos="1" name="hw.product"/>
+    <param pos="2" name="hw.version"/>
+  </fingerprint>
+  <fingerprint pattern="^(F6\d+\w?)\r\n\rLogin:\s*$">
+    <description>ZTE F6xx series GPON router</description>
+    <!-- F668\r\n\rLogin: -->
+    <example _encoding="base64" hw.product="F668">RjY2OA0KDUxvZ2luOgo=</example>
+    <!-- F612W\r\n\rLogin: -->
+    <example _encoding="base64" hw.product="F612W">RjYxMlcNCg1Mb2dpbjoK</example>
+    <param pos="0" name="service.vendor" value="ZTE"/>
+    <param pos="0" name="service.product" value="telnet"/>
```

Comment:
I'm not sure we should be using `telnet` as the `product` here.  Yes, the service is telnet but the rest of the `service` fingerprint implies that ZTE wrote this telnet product, when in reality they (or others) are just using some open source/linux/GNU telnetd implementation.   

Looked at from another perspective, to even perform the recog fingerprinting, one must have already identified that the protocol is telnet and spoken it to obtain the banner.

Reply:
Agreed, I'd been on the fence on this and was likely going to remove it.  We already get back  `"service.protocol":"telnet"` automatically for matches in this file where that value isn't overridden.